### PR TITLE
simplify-boolean-expression is now emitted for old style ternaries with first value being falsy in boolean context

### LINKED ISSUE #1559: consider-using-ternary suggests non-equivalent code
### Steps to reproduce
1. Create a file called to_check.py:
```
"""Test for consider-using-ternary correctness."""
from itertools import product
for condition, true_value, false_value in product([False, True], repeat=3):
    old_value = condition and true_value or false_value
    new_value = true_value if condition else false_value
    if old_value != new_value:
        print(
            '{old_value} == '
            '({condition} and {true_value} or {false_value}) != '
            '({true_value} if {condition} else {false_value}) == '
            '{new_value}'.format(
                condition=condition,
                true_value=true_value,
                false_value=false_value,
                old_value=old_value,
                new_value=new_value,
            ))
```
2. Run `pylint to_check.py`
3. Run `python3 to_check.py`

### Current behavior
Pylint suggests
```
R:  4, 4: Consider using ternary (true_value if condition else false_value) (consider-using-ternary)
```
but the code output shows that this is not an equivalent replacement:
```
True == (True and False or True) != (False if True else True) == False
```

### Expected behavior
Pylint suggests an equivalent replacement.

### pylint --version output
pylint 1.7.2, 
astroid 1.5.3
Python 3.4.3 (default, Nov 17 2016, 01:08:31) 
[GCC 4.8.4]

### PR DESCRIPTION
**Pull request by rogalski**
Close #1559

### CODE REVIEW

**Comment by spiow** [comment]
Hi, I'm the requester of #1559. I'm completely unfamiliar with Pylint internals, so sorry for stupid questions.

I see you've added a check for
```
SOME_VALUE9 = SOME_VALUE1 and False or SOME_VALUE2
```
but will it work the same for a variable:
```
SOME_VALUE_X = False
SOME_VALUE9 = SOME_VALUE1 and SOME_VALUE_X or SOME_VALUE2
```
or a variable not known at compile time:
```
SOME_VALUE_X = some_random_choice([True, False])
SOME_VALUE9 = SOME_VALUE1 and SOME_VALUE_X or SOME_VALUE2
```

**Comment by PCManticore** [comment]
As Sylvain mentioned in the issue, in this case we should just warn that the equivalent version from ``X or False and Y`` is just ``Y``. I know it's a bit hard to do, since we have to modify the code a bit, but it will be more intuitive than skipping the check entirely.

File: ChangeLog [hunk 1]
```diff
@@ -2193,7 +2200,7 @@ Release date: 2009-12-18
 
     * include James Lingard string format checker
 
-    * include simple message (ids) listing by Vincent Férotin (#9791)
+    * include simple message (ids) listing by Vincent F?rotin (#9791)
```
Comment (rogalski) [inline ChangeLog]: Of course I'll revert it, I'm sorry about that

File: ChangeLog [hunk 2]
```diff
@@ -2216,7 +2223,7 @@ Release date: 2009-12-18
 
     * fix test/fulltest.sh
 
-    * #5821 added a utility function to run pylint in another process (patch provide by Vincent Férotin)
+    * #5821 added a utility function to run pylint in another process (patch provide by Vincent F?rotin)
```
Comment (rogalski) [inline ChangeLog]: Same as above

File: pylint/checkers/refactoring.py [hunk 3]
```diff
@@ -108,6 +105,9 @@ class RefactoringChecker(checkers.BaseTokenChecker):
                   'a generator may lead to hard to find bugs. This PEP specify that '
                   'raise StopIteration has to be replaced by a simple return statement',
                   {'minversion': (3, 0)}),
+        'R1709': ("Consider using ternary (%s)",
```
Comment (rogalski) [inline pylint/checkers/refactoring.py]: it does not need to be moved, I'll restore original position

File: pylint/checkers/refactoring.py [hunk 4]
```diff
@@ -438,11 +438,17 @@ def visit_assign(self, node):
         else:
             return
 
-        self.add_message(
-            'consider-using-ternary', node=node,
-            args=(truth_value.as_string(),
-                  cond.as_string(),
-                  false_value.as_string()),)
+        if truth_value.bool_value() is False:
+            # TBD: different message? What would be a good name?
```
Comment (rogalski) [inline pylint/checkers/refactoring.py]: I'd need an insightful input on this. 

What should be content of warning message in this case?
What should be message symbol for both those cases? Same? Different?

File: pylint/checkers/refactoring.py [hunk 5]
```diff
@@ -452,6 +458,7 @@ def _is_and_or_ternary(node):
         Returns true if node is 'condition and true_value else false_value' form.
 
         All of: condition, true_value and false_value should not be a complex boolean expression
+        true_value should not be false in boolean context
```
Comment (rogalski) [inline pylint/checkers/refactoring.py]: this is a lie and needs to be removed

File: pylint/test/functional/ternary.txt [hunk 6]
```diff
@@ -5,3 +5,4 @@ consider-using-ternary:21:func3:Consider using ternary (true_value if condition
 consider-using-ternary:24::Consider using ternary ('ERROR' if some_callable(condition) else 'SUCCESS'):HIGH
 consider-using-ternary:25::Consider using ternary ('greater' if SOME_VALUE1 > 3 else 'not greater'):HIGH
 consider-using-ternary:26::Consider using ternary ('both' if (SOME_VALUE2 > 4) and (SOME_VALUE3) else 'not'):HIGH
+consider-using-ternary:29::Consider using ternary (SOME_VALUE2):HIGH
```
Comment (rogalski) [inline pylint/test/functional/ternary.txt]: Currently it reads quitee ugly. That's why I want to ask for suggestions.

Comment (AWhetter) [inline pylint/test/functional/ternary.txt]: How about "Ternary can be simplified (SOME_VALUE2)", "Ternary can be replaced with (SOME_VALUE2)"?

File: pylint/checkers/refactoring.py (hunk 4 above)
Comment (PCManticore) [inline pylint/checkers/refactoring.py]: ``simplify-boolean-expression`` maybe?

**Comment by PCManticore** [comment]
I'm happy with where this is going @rogalski So once you come up with a new message, feel free to merge this.

File: pylint/test/functional/ternary.txt [hunk 7]
```diff
@@ -5,3 +5,4 @@ consider-using-ternary:21:func3:Consider using ternary (true_value if condition
 consider-using-ternary:24::Consider using ternary ('ERROR' if some_callable(condition) else 'SUCCESS'):HIGH
 consider-using-ternary:25::Consider using ternary ('greater' if SOME_VALUE1 > 3 else 'not greater'):HIGH
 consider-using-ternary:26::Consider using ternary ('both' if (SOME_VALUE2 > 4) and (SOME_VALUE3) else 'not'):HIGH
+simplify-boolean-expression:29::Boolean expression may be simplified (SOME_VALUE2):HIGH
```
Comment (PCManticore) [inline pylint/test/functional/ternary.txt]: How about `may be simplified to: (SOME_VALUE2)`?

**Pull request merged**